File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.cpp
class Solution {
public:
    string smallestPalindrome(string s) {
         int n = s.size();
        int mid = n / 2;

        sort(s.begin(), s.begin() + mid);

        string ans;

        for (int i = 0; i < mid; i++) {
            ans += s[i];
        }

        if (n % 2 == 1) {
            ans += s[mid];
        }

        for (int i = mid - 1; i >= 0; i--) {
            ans += s[i];
        }

        return ans;
    }
};

File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.java
class Solution {
    public String smallestPalindrome(String s) {
        int n = s.length();
        int mid = n / 2;
        char[] arr = s.toCharArray();

        Arrays.sort(arr, 0, mid);

        StringBuilder str = new StringBuilder();
        for(int i = 0; i < mid; i++){
            str.append(arr[i]);
        }

        if (n % 2 == 1) {
            str.append(arr[mid]);
        }

        for(int i = mid - 1; i >= 0; i--){
            str.append(arr[i]);
        }

        return str.toString();
    }
}